### ISSUE TITLE
Any tips and tricks to up the performance?

### ISSUE THREAD

**Issue by lchunleo** [lvapeab/nmt-keras#56]
Hi 

 i had managed to setup and run the NMT. However, the result does not reflect well. I am using ~20K of training data for each of the language corpus and 3K for the evaluation data when using the default config out of the box running for 20 epoches. Any tips to improve the translation acurracy? the translated sentences are quite often too different from the original text in terms of context. 

 i also noticed that there are a few measurements score such as BLEU1--4, TER, ROUGE etc. May i know what each represent?

**Comment by lvapeab** [comment]
Hi, 

indeed, the hyperparameters that are in the `config.py` file aren't probably the optimal for any task. The choice of hyperparameters greatly depends on the task. I can recommend the following changes (but probably other setups are valid too): 

- Change your learning rate. You'll probably need to tune it (try e.g. `LR = 0.0002`).

- Increment the size of your network: `SOURCE_TEXT_EMBEDDING_SIZE`,  `TARGET_TEXT_EMBEDDING_SIZE`,  `ENCODER_HIDDEN_SIZE` and `DECODER_HIDDEN_SIZE` (try values of 256 or higher).

- Try to inject some noise to the weights during training (`USE_NOISE = True`)

- Try to apply dropout: `DROPOUT_P = 0.1` and `ATTENTION_DROPOUT_P = 0.1 ` and a small smoothing on labels `LABEL_SMOOTHING = 0.05`

- The application of [byte-pair-encoding](https://github.com/rsennrich/subword-nmt/) is also recommended. 

For knowing more about the scores reported during training, please follow the references appearing in  [coco-caption](https://github.com/lvapeab/coco-caption/). There you'll got links to those metrics.
